modin/core/storage_formats/base/query_compiler.py

```diff
@@ -1769,24 +1769,33 @@ def to_timedelta(self, unit="ns", errors="raise"):  # noqa: PR02
             self, unit=unit, errors=errors
         )
 
-    # FIXME: get rid of `**kwargs` parameter (Modin issue #3108).
-    @doc_utils.add_one_column_warning
-    @doc_utils.add_refer_to("Series.unique")
-    def unique(self, **kwargs):
+    @doc_utils.add_refer_to("Series.drop_duplicates")
+    def unique(self, keep="first", ignore_index=False, subset=None):
         """
-        Get unique values of `self`.
+        Get unique rows of `self`.
 
         Parameters
         ----------
-        **kwargs : dict
-            Serves compatibility purpose. Does not affect the result.
+        keep : {"first", "last", False}, default: "first"
+            Which duplicates to keep.
+        ignore_index : bool, default: False
+            If ``True``, the resulting axis will be labeled ``0, 1, …, n - 1``.
+        subset : list, optional
+            Only consider certain columns for identifying duplicates, if `None`, use all of the columns.
 
         Returns
         -------
         BaseQueryCompiler
             New QueryCompiler with unique values.
         """
-        return SeriesDefault.register(pandas.Series.unique)(self, **kwargs)
+        if subset is not None:
+            mask = self.getitem_column_array(subset, ignore_order=True)
+        else:
+            mask = self
+        without_duplicates = self.getitem_array(mask.duplicated(keep=keep).invert())
+        if ignore_index:
+            without_duplicates.index = pandas.RangeIndex(len(without_duplicates.index))
```

Comment:
```suggestion
            without_duplicates.index = pandas.RangeIndex(len(without_duplicates))
```

Reply:
good note, but here we're operation on a query compiler that doesn't have the `.__len__()` method, so the only way to get its length is by using its index

Reply:
so is there a point in using this line as the index would have to be computed the set the index? can this be removed?

Reply:
indeed, we can use `.reset_index()` here instead that would reset the index without computing the old one